### PR TITLE
Fixes #22437: Missing API documentation for campaigns

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://issues.rudder.io/issues/22437

### CODE REVIEW

File: webapp/sources/api-doc/code_samples/curl/campaigns/allCampaigns.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+yo
```
Comment (fanf) [inline webapp/sources/api-doc/code_samples/curl/campaigns/allCampaigns.sh:1]: I think there was a bad copy/paste or something

Comment (VinceMacBuche) [reply webapp/sources/api-doc/code_samples/curl/campaigns/allCampaigns.sh:1]: It's part of 'not being finished'

File: webapp/sources/api-doc/paths/campaigns/getEvent.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+# SPDX-FileCopyrightText: 2013-2020 Normation SAS
+get:
+  summary: Get a campaign event details
+  description: Get a campaign event details
+  operationId: getCampaignEvent
+  parameters:
+    - $ref: ../../components/parameters/campaign-event-id-path.yml
+  responses:
+    "200":
+      description: Campaign details result
```
Comment (fanf) [inline webapp/sources/api-doc/paths/campaigns/getEvent.yml:11]: + event

File: webapp/sources/api-doc/paths/campaigns/getEventsCampaign.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+# SPDX-FileCopyrightText: 2013-2020 Normation SAS
+get:
+  summary: Get campaign events for a campaign
+  description: Get campaign events for a campaign
+  operationId: getEventsCampaign
+  parameters:
+    - $ref: ../../components/parameters/campaign-id-path.yml
+  responses:
+    "200":
+      description: Campaign details result
```
Comment (fanf) [inline webapp/sources/api-doc/paths/campaigns/getEventsCampaign.yml:11]: Campaing event details list

File: webapp/sources/api-doc/paths/campaigns/saveEvent.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
+# SPDX-License-Identifier: CC-BY-SA-2.0
+# SPDX-FileCopyrightText: 2013-2020 Normation SAS
+post:
+  summary: Get a campaign event details
```
Comment (fanf) [inline webapp/sources/api-doc/paths/campaigns/saveEvent.yml:4]: Save a campaign event

File: webapp/sources/api-doc/paths/campaigns/saveEvent.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  description: Get a campaign event details
```
Comment (fanf) [inline webapp/sources/api-doc/paths/campaigns/saveEvent.yml:5]: I'm not sure about that one. Does it save the future event based on an existing campaign ?

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

File: webapp/sources/api-doc/components/schemas/campaign-details.yml [hunk 6]
```diff
@@ -0,0 +1,44 @@
+# SPDX-License-Identifier: CC-BY-SA-2.0
+# SPDX-FileCopyrightText: 2013-2020 Normation SAS
```
Comment (ElaadF) [inline webapp/sources/api-doc/components/schemas/campaign-details.yml:2]: 2013-2023

File: webapp/sources/api-doc/components/schemas/campaign-event-details.yml [hunk 7]
```diff
@@ -0,0 +1,41 @@
+# SPDX-License-Identifier: CC-BY-SA-2.0
+# SPDX-FileCopyrightText: 2013-2020 Normation SAS
```
Comment (ElaadF) [inline webapp/sources/api-doc/components/schemas/campaign-event-details.yml:2]: 2013-2023 (there are more

**Comment by amousset** [comment]
OK, merging this PR

**Pull request merged**